packages/flutter_tools/test/commands.shard/hermetic/run_test.dart

```diff
@@ -431,6 +431,32 @@ void main() {
         Cache: () => Cache.test(processManager: FakeProcessManager.any()),
       });
 
+      testUsingContext('fails when --flavor is used with an unsupported target platform', () async {
+        const List<String> runCommand = <String>[
+          'run',
+          '--no-pub',
+          '--no-hot',
+          '--flavor=vanilla'
+        ];
+
+        // Useful for test readability.
+        // ignore: avoid_redundant_argument_values
+        final FakeDevice device = FakeDevice(supportsFlavors: false);
+        testDeviceManager.devices = <Device>[device];
```

Comment:
can we have two devices, one of which IS supported, and another that is NOT supported, and then pass the `--all` flag? I'm guessing in that case we should throw tool exit, right?